### PR TITLE
Ref/feature/keep faulty save data

### PR DESCRIPTION
**Pull request by cblech**

### CODE REVIEW

File: Assets/Scripts/EditorState/SceneFile.cs [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
+using System.Collections.Generic;
 using Assets.Scripts.SceneState;
 using JetBrains.Annotations;
 using UnityEngine;
 
+
 namespace Assets.Scripts.EditorState
 {
-    public class SceneState
+    public class SceneFile
```
Comment (tcdrop) [inline Assets/Scripts/EditorState/SceneFile.cs]: Please rename to something like "SceneDirectoryState" as this class represents the state of the scene directory on the file system.

File: Assets/Scripts/System/SceneLoadSaveSystem.cs [hunk 2]
```diff
@@ -30,65 +42,88 @@ public SceneLoadSaveSystem(PathState pathState)
...
             string sceneDirPath = $"{_pathState.ProjectPath}/dcl-edit/saves/v2/{sceneData.name}.dclscene";
             DirectoryInfo sceneDir = Directory.CreateDirectory(sceneDirPath);
 
-            // Clear scene directory
-            foreach (FileInfo file in sceneDir.GetFiles()) file.Delete();
+            // Clear scene directory from files, that are regenerated
+            foreach (FileInfo file in sceneDir
+                         .GetFiles()
+                         .Where(f => sceneFile.LoadedFilePathsInScene.Contains(f.Name)))
+                // Only delete files that where loaded into the scene.
```
Comment (tcdrop) [inline Assets/Scripts/System/SceneLoadSaveSystem.cs]: typo *were*

File: Assets/Scripts/System/SceneLoadSaveSystem.cs [hunk 3]
```diff
@@ -137,7 +170,19 @@ public DclEntityData(DclEntity entity)
...
+
+                if (guid == default)
+                {
+                    throw new SceneLoadException($"Guid was not set for entity {customName}");
+                }
+
+
+                var dclEntity = new DclEntity(guid, customName, parentGuid ?? default);
 
                 // ReSharper disable once ForeachCanBePartlyConvertedToQueryUsingAnotherGetEnumerator
```
Comment (tcdrop) [inline Assets/Scripts/System/SceneLoadSaveSystem.cs:187]: Is this comment necessary?

File: Assets/Scripts/SceneState/DclScene.cs [hunk 4]
```diff
@@ -21,7 +21,20 @@ public DclEntity GetEntityFormId(Guid id)
             if (id == Guid.Empty)
```
Comment (tcdrop) [inline Assets/Scripts/SceneState/DclScene.cs:20]: A better method name would be "GetEntityById"

**Review by tcdrop — CHANGES_REQUESTED** [review]

File: Assets/Scripts/System/SceneLoadSaveSystem.cs (hunk 3 above)
Comment (cblech) [reply Assets/Scripts/System/SceneLoadSaveSystem.cs:187]: Yes :D
At least I'd prefer to keep it, because ReSharper shows an unwanted warning there

**Review by tcdrop — APPROVED** [review]

**Pull request merged**